src/ui/marker.js

```diff
@@ -604,4 +607,31 @@ export default class Marker extends Evented {
     getPitchAlignment() {
         return this._pitchAlignment;
     }
+
+    /**
+     * Sets the color of the Marker if using a default marker.
+     * @param {string} [color='#3FB1CE'] The color as a string to use for the default marker. If `color` is not provided the default is light blue.
+     * @returns {Marker} `this`
+     */
+    setColor(color: ?string) {
+        if (typeof this._background !== 'undefined') {
```

Comment:
Hey, thanks for your contribution! The way this is set up right now, it would be impossible to set the color of a custom element which limits the usefulness of this method. I think it may be difficult to accomplish this with any arbitrary HTML element because we won't know what element to target to update the color. We'd probably need to take in an optional `element` or `target` argument to know what part of the element is the background. And what happens if the element is namespaced in the developer's marker (e.g. what if the first argument to `setAttributeNS` isn't `null`?) I think we'd need to address these issues before we could accept this PR.

Reply:
@ryanhamley I'd assume setColor would only work if you didn't pass your own custom marker element. If you pass your own element then customising it later is 100% your responsibility, not setMarker's responsibility. That's how I see it.

Reply:
What about `setColor(color, element)`? Do we have any other API methods that only work for default settings? It seems odd to me

Reply:
Because the color option when you create a new marker only works with the default Marker element, so a get set would only need to work when using the default marker. I think it would be too complicated to have color work for a custom marker element and would break the boundary of what the developer controls and what the Marker class controls.